### PR TITLE
refactor(demo): Overhaul examples

### PR DESCRIPTION
**Pull request by rohanbayya**
- Redesign demo app/examples to use Hyperview navigation, 11ty macros and improved file structure.
- Publishes the demo app on GitHub pages.
- Integrates following PRs:
  - #881
  - #884
  - #908
  - #909
  - #904
  - #913
  - #906
  - #915
  - #911
  - #917
  - #919
  - #918
  - #921
  - #922
  - #925

### CODE REVIEW

File: .eleventyignore [hunk 1]
```diff
@@ -1,2 +1,9 @@
-examples/case_studies/contacts/list.xml.njk
-examples/case_studies/contacts/contacts.xml.njk
+examples_old/case_studies/contacts/list.xml.njk
+examples_old/case_studies/contacts/contacts.xml.njk
```
Comment (flochtililoch) [inline .eleventyignore:2]: remove these paths

File: .gitignore [hunk 2]
```diff
@@ -11,6 +11,7 @@ lib/
 storybook/app-key.gen.js
 
 _examples_site/
```
Comment (flochtililoch) [inline .gitignore:13]: I think this directory no longer exists?

File: demo/package.json [hunk 3]
```diff
@@ -2,11 +2,15 @@
...
+  "homepage": ".",
   "scripts": {
     "start": "expo start",
     "android": "expo start --android",
     "ios": "expo start --ios",
-    "web": "expo start --web"
+    "web": "expo start --web",
+    "deploy": "gh-pages -d hyperview",
+    "predeploy": "BASE_URL=https://instawork.github.io/hyperview/public expo export:web ; mv web-build hyperview; cd .. ; npx eleventy",
+    "eject": "expo eject"
```
Comment (flochtililoch) [inline demo/package.json:13]: remove this line

File: package.json [hunk 4]
```diff
@@ -40,7 +40,7 @@
...
     "test:xmlserver": "eleventy --serve --port=8085",
-    "test:validate-xml": "eleventy; find _examples_site test/schema -name '*.xml' -not -path '*/advanced_behaviors/instawork.xml' -not -path '*/advanced_behaviors/custom_share.xml' -not -path '*/advanced_behaviors/alert/behavior.xml' -not -path '*/case_studies/business_rating.xml' -not -path '*/case_studies/_rating_save.xml' -not -path '*/index.xml' -not -path '*/index_hyperview.xml' | xargs xmlschema-validate --schema schema/hyperview.xsd --version 1.1",
+    "test:validate-xml": "eleventy; find demo/hyperview/public test/schema -name '*.xml' -not -path '*/advanced/scroll/*.xml' -not -path '*/advanced_behaviors/instawork.xml' -not -path '*/advanced_behaviors/custom_share.xml' -not -path '*/advanced_behaviors/alert/behavior.xml' -not -path '*/case_studies/business_rating.xml' -not -path '*/case-studies/business-rating/index.xml' -not -path '*/case-studies/business-rating/rating_save.xml'  -not -path '*/case_studies/_rating_save.xml' -not -path '*/case-studies/photos/comments.xml' -not -path '*/index.xml' -not -path '*/index_hyperview.xml' | xargs xmlschema-validate --schema schema/hyperview.xsd --version 1.1",
```
Comment (flochtililoch) [inline package.json:43]: we should clean up the list of `-not -path` to remove files that no longer exist (because their location has moved)

Comment (rohanbayya) [reply package.json:43]: addressing this in another PR: https://github.com/Instawork/hyperview/pull/927

**Review by flochtililoch — APPROVED** [review]
🚀

**Pull request merged**